scipy/stats/_stats.pyx

```diff
@@ -753,3 +753,86 @@ def gaussian_kernel_estimate(points, values, xi, precision, dtype, real _=0):
 
     return np.asarray(estimate)
+
+
+@cython.wraparound(False)
+@cython.boundscheck(False)
+def gaussian_kernel_estimate_log(points, values, xi, precision, dtype, real _=0):
+    """
+    def gaussian_kernel_estimate_log(points, real[:, :] values, xi, precision)
+    Evaluate the log of the estimated pdf on a provided set of points.
+    Parameters
+    ----------
+    points : array_like with shape (n, d)
+        Data points to estimate from in ``d`` dimensions.
+    values : real[:, :] with shape (n, p)
+        Multivariate values associated with the data points.
+    xi : array_like with shape (m, d)
+        Coordinates to evaluate the estimate at in ``d`` dimensions.
+    precision : array_like with shape (d, d)
+        Precision matrix for the Gaussian kernel.
+    Returns
+    -------
+    estimate : double[:, :] with shape (m, p)
+        The log of the multivariate Gaussian kernel estimate evaluated at the input coordinates.
+    """
+    cdef:
+        real[:, :] points_, xi_, values_, log_values_
+        real[:, :] whitening, estimate, estimate_, zeros_estimate_
+        real[:] max_, zeros_max
+        int i, j, k
+        int n, d, m, p
+        real arg, residual, log_arg, log_norm
+
+    n = points.shape[0]
+    d = points.shape[1]
+    m = xi.shape[0]
+    p = values.shape[1]
+
+    # Rescale the data
+    whitening = np.linalg.cholesky(precision).astype(dtype, copy=False)
+    points_ = np.dot(points, whitening).astype(dtype, copy=False)
+    xi_ = np.dot(xi, whitening).astype(dtype, copy=False)
+    values_ = values.astype(dtype, copy=False)
+
+    # Log transformation of values_. It is faster than np.log(values_).
+    log_values_ = np.empty((n, p), dtype)
+    for i in range(n):
+        for k in range(p):
+            log_values_[i, k] = math.log(values_[i, k])
+
+    # Evaluate the normalisation
+    log_norm = (- d / 2) * math.log(2 * PI)
+    for i in range(d):
+        log_norm += math.log(whitening[i, i])
+
+    # Create the result array and evaluate the weighted sum
+    estimate = np.zeros((m, p), dtype)
+    zeros_max = np.zeros((p,), dtype) - np.inf
+    empty_estimate_ = np.empty((p, n), dtype)
+    for j in range(m):
+        max_ = zeros_max.copy()
+        estimate_ = empty_estimate_.copy()
+        for i in range(n):
+            arg = 0
+            log_arg = 0
+            for k in range(d):
+                residual = (points_[i, k] - xi_[j, k])
+                arg += residual * residual
+
+            log_arg = -arg / 2 + log_norm
+            for k in range(p):
+                estimate_[k, i] = log_values_[i, k] + log_arg
+
+                # Collect max value for logsumexp
+                if max_[k] < estimate_[k, i]:
+                    max_[k] = estimate_[k, i]
+
+        # Calculate logsumexp
+        for k in range(p):
+            for i in range(n):
+                estimate[j, k] += math.exp(estimate_[k, i] - max_[k])
+            estimate[j, k] = math.log(estimate[j, k])
```

Comment:
Thank you for checking the code in details. However, I don't agree with you because 
`estimate[j, k] = math.log(estimate[j, k])` don't override the loop.
In the for loop, `math.exp(estimate_[k, i] - max_[k])` for each `i` is added to `estimate[j, k]`
The next line runs log transformation of this result.

If this line was `estimate[j, k] = math.log(estimate_[j, k])`, this would override the loop.

Correct me if I am wrong...

Reply:
Oh, yes I read this too fast. Sorry for the noise you're right.